### PR TITLE
Blogpost about Spark+Hawkular.

### PR DESCRIPTION
**Pull request by jkremser**

### CODE REVIEW

**Comment by abonas** [comment]
is it ok to copy the spark icon here?

**Comment by pilhuhn** [comment]
You should start the article with a motivation about what you are trying to do.
Try to include a diagram of how things are "wired"

**Comment by mtho11** [comment]
Nice Jirka!

File: src/main/jbake/content/blog/2016/08/31/spark-metric-analysis.adoc [hunk 1]
```diff
@@ -0,0 +1,141 @@
+= Metric Data Analysis using the Apache Spark
+Jiri Kremser
+2016-8-31
+:jbake-type: post
+:jbake-status: published
+:jbake-tags: blog, spark, metrics, cassandra
+
+The goal of this blog post is to show how to easily connect link:https://spark.apache.org/[Apache Spark] to the
+Cassandra with the metric data and do some data transformations/analysis.
+
```
Comment (theute) [inline src/main/jbake/content/blog/2016/08/31/spark-metric-analysis.adoc:12]: Agreed with Heiko, a little bit more info into what you will really do here, would be more appealing to read the rest of the article. When I read "do some data transformations/analysis", I don't even have an idea what I would want to do, give me some idea, like the memory correlation (and what that means)

File: src/main/jbake/content/blog/2016/08/31/spark-metric-analysis.adoc [hunk 2]
```diff
@@ -0,0 +1,141 @@
...
+It is very simple, the only assumption here is that the `spark-cassandra-connector` jar is on the class path.
+
+NOTE: Apache Spark is completely written in Scala so also the examples are written in Scala.
+Although, it provides also Java and Python APIs, using the language it has been written in is the best choice,
+because one can track down the calls in the (native) source code.
+
+== Reading the data from Cassandra
+
+Let's create an RDD from Cassandra table called "data" in "hawkular_metrics" keyspace. RDD stands for Resilient
+Distributed Datasets and has link:https://spark.apache.org/docs/latest/api/scala/#org.apache.spark.rdd.RDD[plenty] of
```
Comment (theute) [inline src/main/jbake/content/blog/2016/08/31/spark-metric-analysis.adoc]: "Let's create a "Resilient Distributed Dataset" (RDD) from Cassandra table called "data" in "hawkular_metrics" keyspace. Resilient Distributed Datasets have link:https://spark.apache.org/docs/latest/api/scala/#org.apache.spark.rdd.RDD[plenty] of useful operations defined."

Then you can use RDD, but the first occurence should not be the acronym.

File: src/main/jbake/content/blog/2016/08/31/spark-metric-analysis.adoc [hunk 3]
```diff
@@ -0,0 +1,141 @@
...
+
+[source,scala]
+----
+val correlation: Double = Statistics.corr(used, free, "pearson")
+println(s"Correlation is: $correlation")
+----
+
+This will print `Correlation is: -0.9999999999999781`. That means there is total negative correlation between the two.
+This example is pretty trivial, but in general, one can calculate the correlation between any two metric streams this way.
+
```
Comment (theute) [inline src/main/jbake/content/blog/2016/08/31/spark-metric-analysis.adoc]: A little explanation on how to interpret the number would not be superfluous

**Comment by theute** [comment]
Thanks for the blog, very nice !
I added some comments as I read through, hopefully to be more attractive and clear. (Those are comments from someone who knows nothing about Spark/RDDs, but could maybe dig into it if he sees and understand the results)

**Comment by burmanm** [comment]
I like the idea, I was going to write one myself also, but then I stopped because .. this is going to break in metrics 0.20.0. You can't request data directly from Cassandra as the datapoints will be compressed in the rows and can't be parsed this way. 

We would need a Spark connector to make this work again after that..

**Comment by jkremser** [comment]
> is it ok to copy the spark icon here?

@abonas I hope so, IANAL
https://commons.wikimedia.org/wiki/File:Spark-logo-192x100px.png

**Comment by jkremser** [comment]
> You should start the article with a motivation about what you are trying to do.
> Try to include a diagram of how things are "wired"

@pilhuhn Dovrebbe essere fatto :spaghetti: :pizza:

**Comment by jkremser** [comment]
@burmanm not sure what was changed in 0.20.0, but if it's just enabling the LZ4 or SNAPPY, the datastax connector can work with that

https://github.com/datastax/spark-cassandra-connector/blob/master/doc/reference.md#cassandra-connection-parameters

Provided, it's your own compression algorithm, then yes, we would need own Spark connector.

**Comment by burmanm** [comment]
@Jiri-Kremser Actually, in 0.20.0 there will be a new column that includes a blob and that blob has compressed datapoints. LZ4/Deflate is already used in HWKMETRICS.

**Pull request merged**